appenders/python/pulse/metrics.py

```diff
@@ -0,0 +1,167 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+# Including Apache 2 license because unix_time_micros method was pulled
+# from Apache Kudu project.
+
+from datetime import datetime
+import atexit
+
+from pytz import utc
+import six
+
+from pulse.batcher import PulseBatcher
+
+
+# noinspection PyPep8Naming
+class MetricWriter(PulseBatcher):
+    """
+    A class which sends metrics to the Pulse Log Collector.
+    """
+
+    def __init__(self, endpoint, table, capacity=1000,
+                 threadCount=1, logger=None):
+        """
+        Initializes a MetricWriter using the provided endpoint, table and
+        optional buffer capacity.
+
+        :param endpoint: The REST API endpoint for the Pulse Log Collector
```

Comment:
What's this formatting style called? Are there tools that can parse it?

Reply:
Are you referring to the docstring format? This is reStructuredText (reST), its used by default by JetBrains and you would use Sphinx to generate the docs. I'm sure there are others that use it as well.

Reply:
pyspark uses sphinx: https://spark.apache.org/docs/latest/api/python/index.html

Reply:
Also, since IntelliJ/Pycharm use it, it enforces typing in the IDE which is nice

Reply:
Nice.